scalding-parquet-scrooge/src/main/scala/com/twitter/scalding/parquet/scrooge/ParquetListFormatRule.scala

```diff
@@ -0,0 +1,262 @@
+package com.twitter.scalding.parquet.scrooge
+
+import java.util
+
+import org.apache.parquet.schema.{GroupType, OriginalType, PrimitiveType, Type}
+import org.slf4j.LoggerFactory
+
+/**
+ * Rule to convert parquet schema of legacy list type to standard one
+ * namely 3-level list structure as recommended in
+ * https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#lists
+ *
+ * More specifically this handles converting from parquet file created by
+ * {{@code org.apache.parquet.thrift.ThriftSchemaConvertVisitor}} which always suffix
+ * list element with "_tuple".
+ */
+private[scrooge] object ParquetListFormatRule extends ParquetCollectionFormatRule {
+
+  private val LOGGER = LoggerFactory.getLogger(getClass)
+
+  def formatForwardCompatibleRepeatedType(repeatedSourceType: Type, repeatedTargetType: Type) = {
+    (
+      findFirstListRule(repeatedSourceType, Source),
+      findFirstListRule(repeatedTargetType, Target)
+    ) match {
+      case (Some(sourceRule), Some(targetRule)) => {
+        if (sourceRule == targetRule) {
+          repeatedSourceType
+        } else {
+          val elementType = sourceRule.elementType(repeatedSourceType)
+          targetRule.createCompliantRepeatedType(
+            typ = elementType,
+            name = elementType.getName,
+            isElementRequired = sourceRule.isElementRequired(repeatedSourceType),
+            originalType = sourceRule.elementOriginalType(repeatedSourceType)
+          )
+        }
+      }
+      case _ => repeatedSourceType
+    }
+  }
+
+  def isGroupList(projection: Type): Boolean = {
```

Comment:
To confirm my understanding:

This is matching on things like 

```
required group my_list (LIST) {
  repeated group list {
    optional binary element (UTF8);
  }
}
```

where `groupProjection` here refers to the top-most type, and the `groupProjection.getFields.get(0).isRepetition(Type.Repetition.REPEATED)` check would apply to the second-level element (`repeated group list`)?

Reply:
yup that's correct on `groupProjection`, and the repeated is the second level.

The part after `repeated` could be different depending on formats eg.
```
required group my_list (LIST) {
  repeated repeated int32 [element|array];
}
```